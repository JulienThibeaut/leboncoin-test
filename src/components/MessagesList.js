import React from 'react'
import messagesList from './messagesList.css'

const MessagesList = props => (
  <ul className="message-list">
    {props.messages.map(msg =>
      <li className="message" key={msg.index}>
        <p>{msg.text}</p><p className="visibility">{msg.visibility ? 'private' : 'public'}</p>
      </li>)}
  </ul>
)

export default MessagesList
