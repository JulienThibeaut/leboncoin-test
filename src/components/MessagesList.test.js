import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import renderer from 'react-test-renderer';
import MessagesList from './MessagesList';

it('should render a todo item ', () => {
  const tree = toJson(shallow(<MessagesList messages={[]} />));
  expect(tree).toMatchSnapshot();
});

it('should match its snapshot with messages', () => {
  const messages = ['leboncoin', 'test', 'test technique'];
  const tree = renderer.create(
    <MessagesList messages={messages} />
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
