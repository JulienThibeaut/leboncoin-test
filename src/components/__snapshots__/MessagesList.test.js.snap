// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match its snapshot with messages 1`] = `
<ul
  className="message-list"
>
  <li
    className="message"
  >
    <p />
    <p
      className="visibility"
    >
      public
    </p>
  </li>
  <li
    className="message"
  >
    <p />
    <p
      className="visibility"
    >
      public
    </p>
  </li>
  <li
    className="message"
  >
    <p />
    <p
      className="visibility"
    >
      public
    </p>
  </li>
</ul>
`;
